sefaria/helper/tests/topic_test.py

```diff
@@ -97,18 +97,19 @@ def actual_author(author_root):
 
 def test_title_and_desc(author_root, actual_author, root_with_self_link, child_of_root_with_self_link, grandchild_of_root_with_self_link):
 	for count, t in enumerate([author_root, actual_author, root_with_self_link, child_of_root_with_self_link, grandchild_of_root_with_self_link]):
-		new_values = {"title": f"new title {count+1}",
-					  "altTitles": {"en": [f"New Alt title {count+1}"], "he": [f"New He Alt Title {count+1}"]},
-					  "heTitle": f"new hebrew title {count+1}", "description": {"en": f"new desc", "he": "new hebrew desc"}}
+		new_values = {"titles": [{"text": f"new title {count+1}", "primary": True, "lang": 'en'},
+					  {"lang": "en",  "text": f"New Alt title {count+1}"}, {"lang": "he", "text": f"New He Alt Title {count+1}"},
+					  {"lang": "he", "text": f"new hebrew title {count+1}", "primary": True}], "description": {"en": f"new desc", "he": "new hebrew desc"}}
 		topic.update_topic(t["topic"], **new_values)
 		assert t["topic"].description == new_values["description"]
-		assert t["topic"].get_primary_title('he') == new_values['heTitle']
-		assert t["topic"].get_titles('en') == [new_values["title"]]+new_values["altTitles"]['en']
+		assert t["topic"].get_primary_title('he') == [title['text'] for title in new_values["titles"] if title["lang"]=='he' and title.get('primary', None)][0]
```

Comment:
let's create variables `en_primary_title` and `he_primary_title` which will be used here and to define `new_values` to avoid this list comprehension.